src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -4797,7 +4798,8 @@ void ClassFileParser::verify_legal_method_modifiers(jint flags,
           // ACC_STRICT, or ACC_SYNCHRONIZED flags set.  No need to
           // check for ACC_FINAL, ACC_NATIVE or ACC_SYNCHRONIZED as
           // those flags are illegal irrespective of ACC_ABSTRACT being set or not.
-          (is_abstract && (is_private || is_static || is_strict))) {
+          (is_abstract &&
+           (is_private || is_static || (major_gte_17 ? false : is_strict)))) {
```

Comment:
Lines 4802 and 4830 would be simpler if changed to:
   (is_private || is_static || (!major_gte_17  && is_strict)))) {